interface/Tools.h

```diff
@@ -74,6 +76,261 @@ MELAAngles HHAnalyzer::getMELAAngles(const LorentzVector &q1, const LorentzVecto
     return angles;
 }
 
+void HHAnalyzer::matchOfflineLepton(const HLTProducer& hlt, HH::Dilepton& dilepton) {
+// Function that tries to match `lepton` with an online object, using a deltaR and a deltaPt cut
+// Returns the index inside the HLTProducer collection, or -1 if no match is found.
+// (Originally taken from https://github.com/blinkseb/TTAnalysis/blob/c2a2d5de3e4281943c19c582afb452b8ef6457f1/plugins/TTAnalyzer.cc#L533)
+    if (leptons[dilepton.ilep1].hlt_already_tried_matching && leptons[dilepton.ilep2].hlt_already_tried_matching) {
```

Comment:
Hmmm I don't know. If at some point we ask for exactly two leptons this problem will go away and as you say this configuration is quite unlikely. On the other hand, since we now apply the id and iso cut upstream, we should not have that many dileptons pairs to match and maybe this check is not a big gain. I would go for an intermediate solution which is to put the `hlt_already_tried_matching` to true, the `hlt_idx` to -1, etc., in case `l1/2_all_indices` is empty.